conformance/lang/expressions/unary-expression/unary_minus_expression.balt

```diff
@@ -0,0 +1,1078 @@
+             where operands are of types that are subtypes of int types.
+Labels: byte, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8, int:Unsigned16, int:Unsigned32,
+        module-type-defn, singleton-type, unary-minus
+
+type Ints 1|2;
+
+function errorFunction(int:Signed8 a, int:Signed16 b, int:Signed32 c,
+                     int:Unsigned8 d, int:Unsigned16 e, int:Unsigned32 f,
+                     byte g, Ints h) {
+    int:Signed8 _ = -a; // @error static type of unary-minus expression with operands of int subtypes is int
+
+    int:Signed16 _ = -b; // @error static type of unary-minus expression with operands of int subtypes is int
+
+    int:Signed32 _ = -c; // @error static type of unary-minus expression with operands of int subtypes is int
+
+    int:Unsigned8 _ = -d; // @error static type of unary-minus expression with operands of int subtypes is int
+
+    int:Unsigned16 _ = -e; // @error static type of unary-minus expression with operands of int subtypes is int
+
+    int:Unsigned32 _ = -f; // @error static type of unary-minus expression with operands of int subtypes is int
+
+    byte _ = -g; // @error static type of unary-minus expression with operands of int subtypes is int
+
+    Ints _ = -h; // @error static type of unary-minus expression with operands of int subtypes is int
+}
+
+Test-Case: error
+Description: Test the static type of the result being the basic type of the operands via invalid assignment,
+             where operands are of types that are optional subtypes of int types.
+Labels: byte, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8, int:Unsigned16, int:Unsigned32,
+        optional-type, module-type-defn, singleton-type, unary-minus
+
+type Ints 1|2;
+
+function errorFunction(int:Signed8? a, int:Signed16? b, int:Signed32? c,
+                     int:Unsigned8? d, int:Unsigned16? e, int:Unsigned32? f,
+                     byte? g, Ints? h) {
+    int:Signed8? _ = -a; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+
+    int:Signed16? _ = -b; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+
+    int:Signed32? _ = -c; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+
+    int:Unsigned8? _ = -d; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+
+    int:Unsigned16? _ = -e; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+
+    int:Unsigned32? _ = -f; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+
+    byte? _ = -g; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+
+    Ints? _ = -h; // @error static type of unary-minus expression with operands of optional int subtypes is int optional
+}
```

Comment:
We also need a test for 

> the static type of the result is the basic type of the static type of the operand, **modified by the usual singleton typing rules**.

We can test it when the operand is a singleton type.
e.g. `-1 v = -1;` should work as the type of operand is singleton 1, but the static type of the result should be singleton -1. (not singleton 1).

Reply:
Static type of the result should be the basic type. We can add a test like this 
`1 v = -1; // incompatible types: expected '1', found 'int'`

Reply:
Quoting from the [singleton-typing ](https://ballerina.io/spec/lang/master/#singleton_typing)
> This means that when all subexpressions of the compound expression have a singleton type, then the static type of E is a singleton subtype of T. 

As described above if the operand is a single type then the result also has to be a singleton type.
Currently, `-1 v = -1;` works. If the static type of the result is int this should not work, right?

For me, the error given for the `1 v = -1;` is wrong. I think it has to be `incompatible types: expected '1', found '-1'`.
You can refer to #902 also.

Reply:
Having second thoughts on the error message. For something like below, the error message is correct. 
`string v = 1; // @error incompatible types: expected 'string', found 'int'`
based on that error message of `1 v = -1` should be consistence to that.

Anyway, `-1 v = -1;` works because we modify the basic type of the result with the singleton typing rules.